packages/web3-validator/src/utils.ts

```diff
@@ -449,20 +449,49 @@ export function uint8ArrayToHexString(uint8Array: Uint8Array): string {
 	return hexString;
 }
 
+// for optimized technique for hex to bytes conversion
+const charCodeMap = {
+	zero: 48,
+	nine: 57,
+	A: 65,
+	F: 70,
+	a: 97,
+	f: 102,
+  } as const
+
+  function charCodeToBase16(char: number) {
+	if (char >= charCodeMap.zero && char <= charCodeMap.nine)
+	  return char - charCodeMap.zero
+	if (char >= charCodeMap.A && char <= charCodeMap.F)
+	  return char - (charCodeMap.A - 10)
+	if (char >= charCodeMap.a && char <= charCodeMap.f)
+	  return char - (charCodeMap.a - 10)
+	return undefined
+  }
+
 export function hexToUint8Array(hex: string): Uint8Array {
-	let value;
+	let offset = 0;
 	if (hex.toLowerCase().startsWith('0x')) {
```

Comment:
I think, it is a bit costy to convert the full string to lowercase.
```suggestion
	if (hex[0] === '0' && (hex[1] === 'x' || hex[1] === 'X')) {
```